domain/schema/model/sql/0019-application.sql

```diff
@@ -200,6 +200,42 @@ CREATE TABLE application_status (
     REFERENCES workload_status_value (id)
 );
 
+CREATE TABLE device_constraint (
+    application_uuid TEXT NOT NULL,
+    name TEXT NOT NULL,
+    type TEXT NOT NULL,
+    count INT,
+    attribute_uuid TEXT,
+    CONSTRAINT fk_device_constraint_application
+    FOREIGN KEY (application_uuid)
+    REFERENCES application (uuid),
+    CONSTRAINT fk_device_constraint_attribute
+    FOREIGN KEY (attribute_uuid)
+    REFERENCES device_constraint_attribute (uuid),
+    PRIMARY KEY (application_uuid, name, attribute_uuid)
+);
+
+CREATE TABLE device_constraint_attribute (
+    uuid TEXT NOT NULL PRIMARY KEY,
+    application_uuid TEXT NOT NULL,
+    "key" TEXT NOT NULL,
+    value TEXT NOT NULL,
+    CONSTRAINT fk_device_constraint_attribute_application
+    FOREIGN KEY (application_uuid)
+    REFERENCES application (uuid)
+);
+
+CREATE VIEW v_device_constraint AS
```

Comment:
I'm not a fan of creating views in _anticipation_ of usage patterns.

A lot of the times we end up with them used in one query.